### ISSUE TITLE
ElixirLS formats files it shouldn't

### ISSUE THREAD

**Issue by doughsay** [elixir-lsp/elixir-ls#275]
### Environment

* Elixir & Erlang versions (elixir --version): Elixir 1.10.2 (compiled with Erlang/OTP 22)
* Operating system: Linux 5.6.15-arch1-1 x86_64
* Editor or IDE name (e.g. Emacs/VSCode): VSCode
* LSP Client name: "ElixirLS: Elixir support and debugger"

### Troubleshooting

- [x] Restart your editor (which will restart ElixirLS) sometimes fixes issues
- [x] Stop your editor, remove the entire `.elixir_ls` directory, then restart your editor

### Logs

1.  If using a client other than VS Code, please try VSCode's "ElixirLS: Elixir support and debugger" extension. Does it reproduce your failure?
2.  Create a new Mix project with `mix new empty`, then open that project with VS Code and open an Elixir file. Is your issue reproducible on the empty project? If not, please publish a repo on Github that does reproduce it.
3.  Check the output log by opening `View > Output` and selecting "ElixirLS" in the dropdown. Please include any output that looks relevant. (If ElixirLS isn't in the dropdown, the server failed to launch.)
4.  Check the developer console by opening `Help > Toggle Developer Tools` and include any errors that look relevant.

No relevant logs...

### Issue

ElixirLS formats file using the elixir code formatter that it seemingly shouldn't.  This wouldn't be that big of an annoyance if it weren't for the formatting of `mix.lock`.

STR:

1. `mix new elixir_ls_test`
2. `cd elixir_ls_test`
3. edit mix.exs and add `{:plug, "~> 1.0"}` to deps (for example)
4. `mix deps.get`
5. add a new file with a simple module in `not_lib/foo.ex`

```elixir
# .formatter.exs
[
  inputs: ["{mix,.formatter}.exs", "{config,lib,test}/**/*.{ex,exs}"]
]
```

Note `mix.lock` is not covered by the above pattern, nor is `not_lib/foo.ex`

* Running `mix format` will NOT format `mix.lock` nor `not_lib/foo.ex`
* With "save on format" turned on in VSCode, opening `mix.lock` or `not_lib/foo.ex` in vscode and saving them will format them using the elixir formatter.

As stated above, the real annoyance is with `mix.lock`.  I frequently fix merge conflicts in this file, and up until the most recent update to ElixirLS, saving mix.lock would not trigger a format of it.  Since the most recent update however it does trigger a format, which causes each dependency to longer be on a single line.  It ends up fighting with the `mix deps` commands (like `mix deps.unlock plug` and `mix deps.get`) which don't format the lock file.

Here's an example of the before and after of the simple mix.lock described by the steps above:

```elixir
# before save
%{
  "mime": {:hex, :mime, "1.3.1", "30ce04ab3175b6ad0bdce0035cba77bba68b813d523d1aac73d9781b4d193cf8", [:mix], [], "hexpm", "6cbe761d6a0ca5a31a0931bf4c63204bceb64538e664a8ecf784a9a6f3b875f1"},
  "plug": {:hex, :plug, "1.10.1", "c56a6d9da7042d581159bcbaef873ba9d87f15dce85420b0d287bca19f40f9bd", [:mix], [{:mime, "~> 1.0", [hex: :mime, repo: "hexpm", optional: false]}, {:plug_crypto, "~> 1.1.1 or ~> 1.2", [hex: :plug_crypto, repo: "hexpm", optional: false]}, {:telemetry, "~> 0.4", [hex: :telemetry, repo: "hexpm", optional: true]}], "hexpm", "b5cd52259817eb8a31f2454912ba1cff4990bca7811918878091cb2ab9e52cb8"},
  "plug_crypto": {:hex, :plug_crypto, "1.1.2", "bdd187572cc26dbd95b87136290425f2b580a116d3fb1f564216918c9730d227", [:mix], [], "hexpm", "6b8b608f895b6ffcfad49c37c7883e8df98ae19c6a28113b02aa1e9c5b22d6b5"},
}
```

```elixir
# after save
%{
  mime:
    {:hex, :mime, "1.3.1", "30ce04ab3175b6ad0bdce0035cba77bba68b813d523d1aac73d9781b4d193cf8",
     [:mix], [], "hexpm", "6cbe761d6a0ca5a31a0931bf4c63204bceb64538e664a8ecf784a9a6f3b875f1"},
  plug:
    {:hex, :plug, "1.10.1", "c56a6d9da7042d581159bcbaef873ba9d87f15dce85420b0d287bca19f40f9bd",
     [:mix],
     [
       {:mime, "~> 1.0", [hex: :mime, repo: "hexpm", optional: false]},
       {:plug_crypto, "~> 1.1.1 or ~> 1.2", [hex: :plug_crypto, repo: "hexpm", optional: false]},
       {:telemetry, "~> 0.4", [hex: :telemetry, repo: "hexpm", optional: true]}
     ], "hexpm", "b5cd52259817eb8a31f2454912ba1cff4990bca7811918878091cb2ab9e52cb8"},
  plug_crypto:
    {:hex, :plug_crypto, "1.1.2",
     "bdd187572cc26dbd95b87136290425f2b580a116d3fb1f564216918c9730d227", [:mix], [], "hexpm",
     "6b8b608f895b6ffcfad49c37c7883e8df98ae19c6a28113b02aa1e9c5b22d6b5"}
}
```

I hope this makes sense, and I do realize it might be counter to what may be expected if directly requesting to format a file (e.g. using the command palette)

**Comment by axelson** [comment]
Hmmm, that is no good. I think there's two potential solutions:
1. Ensure that the VSCode extension does not try to auto-format `mix.lock`
2. Do not format `mix.lock` within ElixirLS

I'm leaning towards option 1, but I'm not sure where that would have changed recently. Do you know which version of the plugin introduced this behavior?

**Comment by doughsay** [comment]
I'm sorry, I don't know the exact versions...  I just recently switched from ElixirLS Fork back to the non-fork, as per your most recent post on Elixir Forums (https://elixirforum.com/t/introducing-elixirls-the-elixir-language-server/5857/96)  So the move from ElixirLS Fork to ElixirLS was what introduced the behavior for me, and that was about a week ago.

**Comment by doughsay** [comment]
Ok, I just confirmed.  I switched back and forth again between "ElixirLS Fork" and "ElixirLS".  The formatting does not happen on the fork, but does happen on the original, so it just appears to be the difference between 0.3.9999 and 0.4.0.

**Comment by axelson** [comment]
Thanks for checking on that. It's not immediately obvious which PR caused the change in behavior, but I should be able to narrow it down from there.

**Comment by keathley** [comment]
I'm pretty sure I've seen a related issue. All of my projects "disable" the formatter by setting the .formatter.exs file like so:

```elixir
[
  inputs: []
]
```

Despite that, I still get PRs that churn these files because the formatter is being run on them. I tested with the latest published vs code and elixir-ls plugin and was able to reproduce this same issue by setting the format on save option to true. I'm not worried about solving this for myself (since I don't use elixir-ls). I don't follow this project so I'm not sure if its an issue between forks or not.

**Comment by jayjun** [comment]
Fixed in #315.
